### DISCUSSION TITLE
Hint class-string of Model and some interface, and its query builder

### DISCUSSION THREAD

**Discussion by rudiedirkx** [larastan/larastan#2149] [Q&A]
I have a class-string of `MyModel&MyTranslatable`, and I want to load some models, and do something translation-related:

```php
$class = $this->getModelClass($content_type); // class-string<MyModel&MyTranslatable>
$local_objects = $class::query()->findMany($ids); // EloquentCollection<int, EloquentModel>
foreach ($local_objects as $object) {
  $object->saveTranslations($locale, $trans); // Call to an undefined method EloquentModel::saveTranslations()
}
```

I know `$local_objects` is really a `ModelCollection<int, MyModel&MyTranslatable>`, but Larastan doesn't understand the complicated `$class` in `$class::query()->findMany()`.

How should I type this better? `getModelClass()` is a specific `MyModel` class that certainly implements `MyTranslatable`, but I don't know which class, because that config is dynamic, so I can't make it `Country|Language|Test|Etc`.

I understand Larastan doesn't get it. It'be amazing if it did. What would even be the type of `$class::query()`!? Magic. Too magic. That's fine. Larastan doesn't have to change. I can change. But how?

For now I've added a

```php
/** @var ModelCollection<int, Model&TranslatableContract> $local_objects */
```

and that works, but I never like `@var`, and I use the `getModelClass()` class all over, so I'll have `@var` all over :-(

**Comment by calebdw** [comment]
You can use `@return` on the getModelClass method?

**Reply by rudiedirkx** [reply]
I thought that goes without saying: `@return class-string<Model&TranslatableContract>`. That's where the class-string comes from. But you can't do a `$classString::query()` or `get()` on that. Larastan doesn't know what that means.

**Reply by calebdw** [reply]
Sounds like a bug

**Reply by rudiedirkx** [reply]
It is kind of a weird thing to hint IMO, so maybe it's just not explicitly implemented, instead of a bug. I don't even know if it should work, or it's just too hacky. I don't know a better way though. That's why I'm here. For you to tell me what I did wrong and to give me the perfect solution that is perfect 😆
